snowdrop-block/src/Snowdrop/Block/StateConfiguration.hs

```diff
@@ -24,42 +24,21 @@ import           Snowdrop.Block.Types (Block (..), BlockRef, BlockUndo,
 --  (including tip block, currently adopted "best" chain),
 --  while state contains actual blockchain state
 --  as result of application of currently adopted "best" chain on initial blockchain state.
-data BlkStateConfiguration blkType m = BlkStateConfiguration
-    { bscApplyPayload :: Payload blkType -> m (BlockUndo blkType)
-    -- ^ Apply block payload to state. Note, that this method encapsulates validation of
-    -- payload (and inidividual transactions contained in it) as well as actual application
-    -- of payload to state.
-    -- Payload application produces an undo object, which may further be used to revert
-    -- the changes of payload application.
-
-    , bscExpand       :: RawBlk blkType -> m (Block (BlockHeader blkType) (Payload blkType))
-    -- ^ Expand raw block
-
-    , bscApplyUndo    :: BlockUndo blkType -> m ()
-    -- ^ Apply undo: revert changes made by application of block,
-    -- which produced the passed undo object.
-
-    , bscRemoveBlund  :: BlockRef blkType -> m ()
-    -- ^ Remove block from block storage
-
-    , bscStoreBlund   :: RawBlund blkType -> m ()
-    -- ^ Store block along with undo in block storage
-
-    , bscGetBlund     :: BlockRef blkType
-                      -> m (Maybe (RawBlund blkType))
-    -- ^ Retrieve block along with undo from block storage
-
-    , bscBlockExists  :: BlockRef blkType -> m Bool
-    -- ^ Check on whether block with given reference exists in block storage
-
-    , bscGetTip       :: m (Maybe (BlockRef blkType))
-    -- ^ Retrieve tip block reference
-    -- (`Nothing` in case of empty chain being a currently adopted "best" chain)
-
-    , bscSetTip       :: Maybe (BlockRef blkType) -> m ()
-    -- ^ Update tip block reference for new adopted "best" chain
-    -- (`Nothing` in case of empty chain)
-    
-    , bscConfig       :: BlkConfiguration blkType
-    -- ^ Configuration for block sequence validation
+data BlkStateConfiguration chgAccum blkType m = BlkStateConfiguration
+    { bscExpandHeaders   :: chgAccum -> [ RawBlk blkType ] -> m ( OldestFirst [] (BlockHeader blkType) )
+    -- ^ Required for step 1
```

Comment:
As in haddock steps are not mentioned, the Haddock shall be rewritten. Either add all steps to the Haddock for the type or inline short descriptions of steps into Haddocks for the fields. Perhaps, both will be beneficial.